Pulsarc/UI/Buttons/NavigationButton.cs

```diff
@@ -25,22 +25,27 @@ class NavigationButton : Drawable
         /// <param name="anchor">The anchor for this button, defaults to Anchor.Center.</param>
         /// <param name="removeFirst">Whether or not the current screen should be removed
         /// before moving to the navigation screen. True = Remove Current screen before navigating.</param>
-        public NavigationButton(PulsarcScreen screen, int type, string text, Vector2 position, Anchor anchor = Anchor.Center, bool removeFirst = false) : base(Skin.assets["button_back_"+type], position, anchor: anchor)
+        public NavigationButton(PulsarcScreen screen, int type, string text, Vector2 position, Anchor anchor = Anchor.Center, bool removeFirst = false, Anchor textAnchor = Anchor.Center, Color? textColor = null, int fontSize = 18)
+            : this(screen, type, position, new TextDisplayElement(text, new Vector2(position.X, position.Y), fontSize, textAnchor, textColor)) {}
+        
+
+        public NavigationButton(PulsarcScreen screen, int type, Vector2 position, TextDisplayElement text, Anchor anchor = Anchor.Center, bool removeFirst = false)
+            : base(Skin.assets["button_back_" + type], position, anchor: anchor)
         {
-            this.text = new TextDisplayElement(text,new Vector2(position.X - (1 - scale)*10, position.Y - (1 - scale) * 10),color: Color.Black, anchor: Anchor.Center);
+            this.text = text;
+            this.text.move(new Vector2((1 - scale) * -10, (1 - scale) * -10)); // TODO: Change positioniong depending on textAnchor | Position text properly, without using hacky workarounds
```

Comment:
Perhaps put `TODO:` on a new line? Hard to read otherwise.

Reply:
Yep, there's plenty of other places where the comments aren't on a new line too. I've been working on some cleanup (haven't pushed it yet), and one of the things I'm doing is putting these comments on new lines. Thanks for pointing out this one!